modules/Feature/Sources/Restaurants/RestaurantDetailsFactory.swift

```diff
@@ -1,19 +1,18 @@
 import UIKit
+import ServicesInterface
 
 public final class RestaurantDetailsFactory {
     public static func make(
-        restaurantName: String,
-        restaurantDetails: String,
+        restaurant: Restaurant,
         delegate: RestaurantActionsDelegate?,
+        deliveryClient: DeliveryClientProtocol,
```

Comment:
Consider another approach that removes this from the parameters... We can discuss it later today.